Single package install test now checks for installed files

sebkuehn opened the pull request:
test now includes checking if test-install-directory is empty and if all files of the foo-package are correctly installed in the install-directory

sebkuehn committed:
test now includes checking if test-install-directory is empty and if all
files of the foo-package are correctly installed in the
install-directory

sebkuehn committed:
New package uninstall testcase. This testcase is still empty.

sebkuehn committed:
extended test case for single package installation

sebkuehn committed:
Merge branch 'develop' of https://github.com/sebkuehn/openthinclient-manager.git into develop

fernanfs commented on services/package-manager/src/test/java/org/openthinclient/pkgmgr/it/PackageInstallTest.java:
```diff
@@ -56,7 +58,44 @@ public void testInstallSinglePackage() throws Exception {
+    assertTrue(bar2Package.isPresent());
+    assertTrue(packageManager.install(Collections.singletonList(bar2Package.get())));
+
+    File fooXml = new File(configuration.getInstallDir().getPath() + File.separator + "schema" + File.separator + "application" + File.separator + "foo.xml");
```
These File-construction lines are not readable. Please change that to something like the following:

```
    final Path installDirectory = configuration.getInstallDir().toPath();

    Path fooXml = installDirectory.resolve("schema").resolve("application").resolve("foo.xml");
```

Whether or not the files exist can be verified using the following:

```
    Files.isRegularFile(fooXml);
    Files.exists(fooXml);
```

This will make the code overall more readable.

fernanfs commented on services/package-manager/src/test/java/org/openthinclient/pkgmgr/it/PackageInstallTest.java:
```diff
@@ -56,7 +58,44 @@ public void testInstallSinglePackage() throws Exception {
+    File bar2Sample = new File(configuration.getInstallDir().getPath() + File.separator + "schema" + File.separator + "application" + File.separator + "bar2-tiny.xml.sample");
+    File bar2Sfs = new File(configuration.getInstallDir().getPath() + File.separator + "sfs" + File.separator + "package" + File.separator + "bar2.sfs");
+
+    assertTrue("foo.xml not installed",fooXml.exists());
```
In conjunction with the comment above, it would be nice to introduce custom assertion Methods which will require less maintenance. 

```
public void assertFileExists(Path path) {
  assertTrue(path.getFileName() + " does not exist", Files.exists(path));
  assertTrue(path.getFileName() + " is not a regular file", Files.isRegularFile(path));
}
```

sebkuehn committed:
changed file construction and introduced a custom assertion method

according to a proposal from francois the file construction is changed
to a more readable way and a custom assertion method is introduced to
check for file status at one single point

sebkuehn committed:
Merge remote-tracking branch 'openthinclient/develop' into develop

update own fork for further development

sebkuehn committed:
PackageUninstallTest runs now a method which installs all requiered packages before the uninstall happens

fernanfs commented on services/package-manager/src/test/java/org/openthinclient/pkgmgr/it/PackageUninstallTest.java:
```diff
@@ -0,0 +1,92 @@
+	  assertEquals(4, packageManager.getInstallablePackages().size());
+	  
+	  return packageManager;
+} nager.getInstalledPackages().stream().filter
```
This line seems to be broken

sebkuehn committed:
fixed broken line by removing by mistake inserted codepiece

sebkuehn committed:
implement a method whicht generates the filepaths of installed packages only with given package name and installation directory

sebkuehn committed:
added missing throws declaration in PackageUninstall-Test-Case

sebkuehn committed:
uncommented lines which where commented for debugging purpose

sebkuehn committed:
added rudimentary methods for clearing and reseting the installation
directories and package-db's

sebkuehn committed:
added additional messages in assertions to identify clearly what went
wrong during test

sebkuehn committed:
Changed creation of the newFile and oldFile object in the
installPackages method to a construction with filename and parent
directory-path. Before they have been partly constructed only by a
generated string which caused in failures deleting directories in the
install-test directory.

sebkuehn committed:
single package installation test checks now the seperate installation of
the two packages foo and zonk

sebkuehn committed:
added more complex methods for reseting package database and
installation-directories. Until now the reseting of the packageManagers
didn't finished correct.
